src/channels/router_http_channel.erl

```diff
@@ -174,64 +182,101 @@ is_non_local_address(Host) ->
-                maps:merge(Result0, #{
-                    debug => maps:merge(Debug, #{
-                        res => #{
-                            code => StatusCode,
-                            headers => ResponseHeaders,
-                            body => <<>>
-                        }
-                    }),
-                    status => success,
-                    description => <<"Connection established">>
-                });
-            {ok, StatusCode, ResponseHeaders, ResponseBody} when
-                StatusCode >= 200, StatusCode =< 300
-            ->
-                router_device_channels_worker:handle_downlink(Pid, ResponseBody, Channel),
-                maps:merge(Result0, #{
-                    debug => maps:merge(Debug, #{
-                        res => #{
-                            code => StatusCode,
-                            headers => ResponseHeaders,
-                            body => ResponseBody
-                        }
-                    }),
-                    status => success,
-                    description => ResponseBody
-                });
-            {ok, StatusCode, ResponseHeaders, ResponseBody} ->
-                SCBin = erlang:integer_to_binary(StatusCode),
-                maps:merge(Result0, #{
-                    debug => maps:merge(Debug, #{
-                        res => #{
-                            code => StatusCode,
-                            headers => ResponseHeaders,
-                            body => ResponseBody
-                        }
-                    }),
-                    status => failure,
-                    description => <<"ResponseCode: ", SCBin/binary, " Body ", ResponseBody/binary>>
-                });
-            {error, Reason} ->
-                maps:merge(Result0, #{
-                    debug => maps:merge(Debug, #{res => #{}}),
-                    status => failure,
-                    description => list_to_binary(io_lib:format("~p", [Reason]))
-                })
-        end,
-    router_device_channels_worker:report_status(Pid, Ref, Result1).
+
+    case Res of
+        {ok, StatusCode, ResponseHeaders, <<>>} when StatusCode >= 200, StatusCode =< 300 ->
+            maps:merge(Result0, #{
+                response => #{
+                    code => StatusCode,
+                    headers => ResponseHeaders,
+                    body => <<>>
+                },
+                status => success,
+                description => <<"Connection established">>
+            });
+        {ok, StatusCode, ResponseHeaders, ResponseBody} when StatusCode >= 200, StatusCode =< 300 ->
+            maps:merge(Result0, #{
+                response => #{
+                    code => StatusCode,
+                    headers => ResponseHeaders,
+                    body => ResponseBody
+                },
+                status => success,
+                description => ResponseBody
```

Comment:
I am not sure about this, what if body is json string do we wanna show that as description?

Reply:
Let's got with generic descriptions for now.